core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableInterpretable.java

```diff
@@ -102,14 +112,61 @@ public static Bindable toBindable(Map<String, Object> parameters,
       if (spark != null && spark.enabled()) {
         return spark.compile(expr, s);
       } else {
-        return getBindable(expr, s, rel.getRowType().getFieldCount());
+        final int fieldCount = rel.getRowType().getFieldCount();
+        CacheCandidateDetector detector = new CacheCandidateDetector();
+        if (detector.isCachable(rel)) {
+          return BINDABLE_CACHE.get(s, () -> getBindable(expr, s, fieldCount));
+        }
+        return getBindable(expr, s, fieldCount);
       }
     } catch (Exception e) {
       throw Helper.INSTANCE.wrap("Error while compiling generated Java code:\n"
           + s, e);
     }
   }
 
+  /***/
```

Comment:
??? comments ?

Reply:
I am planning to remove the CacheCandidateDetector but thanks for noticing.